### DISCUSSION TITLE
Is there a way to get the PDF page size (width and height) in millimeters?

### DISCUSSION THREAD

**Discussion by araichev** [py-pdf/pypdf#1333] [Q&A]
Couldn't find this feature in the docs and wondering if it's possible, because sometimes i want to check if a PDF is size A4 or A3.

**Comment by pubpub-zz** [comment]
@araichev
the mediabox page property returns the page size in user space units (1 user space unit = 1/72" = 0.35277777777777775 mm)

**Comment by MartinThoma** [comment]
First you have to know that PDF pages have several boxes:

* mediabox
* cropbox
* artbox
* trimbox
* bleedbox

You can get those like this:

```python
>>> from PyPDF2 import PdfReader
>>> reader = PdfReader("issue-856.pdf")
>>> reader.pages[0].mediabox
RectangleObject([0, 0, 595.276, 841.89])
```

The specs say:

> (Required; inheritable) A rectangle (see Section 3.8.4, “Rectangles”), expressed in **default user space units**, defining the boundaries of the physical medium on which the page is intended to be displayed or printed (see Section 10.10.1, “Page Boundaries”).

Chapter 4:

> To avoid the device-dependent effects of specifying objects in device space, PDF defines a device-independent coordinate system that always bears the same relationship to the current page, regardless of the output device on which printing or displaying occurs. This device-independent coordinate system is called user space.
> [...]
> The length of a unit along both the x and y axes is set by the **UserUnit entry (PDF 1.6) in the page dictionary** (see Table 3.27). If that entry is not present or supported, the **default value of 1 ⁄ 72 inch** is used. This coordinate system is called default user space.

You will be able to access a pages user_unit after the next release (see https://github.com/py-pdf/PyPDF2/pull/1336 ) more conveniently.

So the example page above has a media-box of a height of 841.89 * (1/72) inch and a width of 595.276 * (1/72) inch.

You could compare against `PyPDF2.PaperSize.A4` which is defined as (595, 842) - always in the default user space units of 1/72 inch.
